### ISSUE TITLE
Suggestion

### ISSUE THREAD

**Issue by GregoryGoncalves** [open-source-society/computer-science#380]
Hello, i've been roughly following the OSS curriculum, however 2 courses bothered me a bit for different reasons.

How to code: Systematic program design just didn't work for me but for personal reasons, since the method of teaching just didn't click, and Algorithm's part 1 and 2 were far too dry for me to keep attention for as long as my usual daily routine of studies.

I replaced Systematic program design by MIT's SICP, and algorithm's is not really replaceable as most other big courses dealing with algorithms are offline, but is a course this heavy a necessity at this point? 
The knowledge is certainly important, but the algorithms course did feel to me as more of a cheat sheet than a course, i felt like there was a bridge missing between giving away the algorithms and moving between each subject.

I've found 2 courses that might be interesting options.

CS61A is along the lines of SICP, it's a course about complexity but instead of Scheme it uses Python and the teacher is pretty good, too. The classes are far more dynamic and interesting, at least they were to me, as these are subjective areas.

CS62 deals with data structures and algorithms, maybe being a good introduction to Algorithms part 1 and 2.

Links for both: 
http://cs61a.org/ 
http://datastructur.es/sp16/

Of course, these are highly subjective opinions, it's just how i felt when following these 2 courses, and these are the replacements i found, i thought it might be helpful to share them.

**Comment by srs1978** [comment]
CS61A is an intro class, similar to CS50.  CS61A is also a nearly direct port from SICP to Python.  I don't understand how in anyway they'd replace Systematic Program Design.  It would replace CS50.  I think CS50 is not as good as MIT's current intro course, but MIT's course has technical MOOC issues and that's understandable.

Algorithms/Data Structures are what separate the really great programmers from the ordinary.  I have personally completed the Sedgewick courses and they are great.  The linked Java course is IMHO lackluster.

**Comment by GregoryGoncalves** [comment]
CS50 doesn't deal with complexity or even the notion of scalability, the approaches might be introductory but the content is pretty different.

**Comment by YasserKa** [comment]
I totally agree with you in the Systematic program design courses. I found it irritating, since I should learn a new language just for the sake of this course, and I didn't find the material covered helpful. I do agree that the instructor in the algorithm courses was a bit dry, but he did recommend to check the [booksite](http://algs4.cs.princeton.edu/21elementary/)  or the [book ](electronic)that the course is based on. For me, books are a better learning source than videos, since it covers more materials and its easier to refer it.

**Comment by wavded** [comment]
Although the systematic design courses aggravated me (at first).  I did find the concepts helpful in the end such that I'm translating some of the course material to other languages like JavaScript: https://strongloop.com/strongblog/systematic-function-design-in-javascript/

I think the teachers had to deal with this interesting mix of new programmers and experienced programmers, hence the use of a teaching focused functional language (I ended up skipping a lot of stuff that was familiar to me).

**Comment by alexerhardt** [comment]
Hi, I wanted to leave some feedback since it seems that I'm at a similar stage as many of you.

Firstly, I also felt that Systematic Program Design was a tough pill to swallow, especially coming from CS50 which is such a fascinating course. I'm going to echo what others have said: I found learning a new language just for SPD very demotivating, and I didn't get much from it. I've found that some of the maxims that the course tries to teach, such as that program design follows data design, are grasped intuitively through the programming practice you see in courses like CS50 and Princeton's Algorithms. I finished SPD I but I'm not going to take parts II and III.

I have to disagree with @GregoryGoncalves on Princeton's Algorithms though - I'm finding it absolutely gripping. The assignments are stellar, and they feel like just the right level of challenge coming from CS50 - there's barely any skeleton code and there's much less hand-holding than CS50 generally, which is a great next step. I'm half-way through and I would definitely say it's made me a better programmer. A note of caution however - the assignments can easily take 20 or 30 hours to finish in my experience, and the OSSU curriculum seems to underestimate that. I'm certainly not particularly fast as I'm still a beginner, but I know from reading the course materials and reading others' experience that this is not uncommon. Maybe that should be updated in the curriculum to be more conservative.

I completely agree with @srs1978 on San Diego's OOP course. The videos are OK, but the programming assignments are very lacklustre - very easy and with tons and tons of skeleton code. It's basically 90% skeleton code with 10% "fill these gaps". It's funny that I've actually learnt more Java and OOP principles from Princeton's Algorithms than this... Maybe there's a better OOP  course out there? At the very least I recommend taking them together like I've been doing. If I had finished Algos I and II before OOP, I would have never finished the latter because of a lack of challenge.

Also, thanks to @GregoryGoncalves for the excellent suggestions. Don't know if they'll make it to the curriculum but I've certainly noted them. 

Finally, is there an active discussion channel for OSSU, such as Slack, Facebook, Discord, etc.? Thanks for letting me know, I'd be happy to join. If it doesn't exist, we should create one!

**Comment by ericdouglas** [comment]
Many thanks for your feedback guys, I really appreciate it. 

I just opened an issue to discuss our next curriculum update [here](https://github.com/open-source-society/computer-science/issues/387), if you have more suggestions, feel free to add them here on the other issue.

**Comment by ericdouglas** [comment]
@alexerhardt we have a [discussion channel](https://gitter.im/open-source-society/computer-science) :smile:

**Comment by VipinindKumar** [comment]
I completed the Systematic Programming Design series recently, and I have to disagree with many of you.

 I was skeptical at first when I started the first course because of learning Racket language for the course and I was highly demotivated. So I did some research and found very convincing arguments in favor of taking the course. So I made the decision of taking the course and I completed it. And it was a great series of courses. I was surprised that following the systematic approach made the complex program so much easier to solve and understand, the same program which would be a real pain to solve without it.

I learned a lot from the series, program design, functional programming, data structures(trees, graphs ...), algorithms.
